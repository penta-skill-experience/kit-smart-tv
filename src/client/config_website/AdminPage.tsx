import * as React from "react";
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import {Grid} from "@mui/material";
import InputLabel from "@mui/material/InputLabel";
import Input from "@mui/material/Input";
import InputAdornment from "@mui/material/InputAdornment";
import IconButton from "@mui/material/IconButton";
import VisibilityOff from "@mui/icons-material/VisibilityOff";
import Visibility from "@mui/icons-material/Visibility";
import FormControl from "@mui/material/FormControl";
import Snackbar from "@mui/material/Snackbar";

export const AdminPage = ({oldPassword, newPassword, handleOldPassword, handleNewPassword, handlePasswordChange, children} ) => {
    const [showOldPassword, setShowOldPassword] = React.useState(false);
    const [showNewPassword, setShowNewPassword] = React.useState(false);

    const handleShowOldPassword  = () => {
        setShowOldPassword(!showOldPassword)
    }

    const handleShowNewPassword  = () => {
        setShowNewPassword(!showNewPassword)
    }

    const [successfulBar, setSuccessfulBar] = React.useState(false);
    const [errorBar, setErrorBar] = React.useState(false);

    const handleClose = (event: React.SyntheticEvent | Event, reason?: string) => {
        if (reason === 'clickaway') {
            return;
        }
        setSuccessfulBar(false);
        setErrorBar(false);
    }

    return (
        <Box
        sx={{
            width: 300,
            height: 300,
            }
        }
        >
            <div>
                <Grid container spacing={2} direction="row" justifyContent="center" alignItems="center">
                    <Grid item xs={12}>
                        <FormControl sx={{ m: 1, width: '25ch' }} variant="standard">
                            <InputLabel>Old Password</InputLabel>
                            <Input
                                type={showOldPassword ? 'text' : 'password'}
                                value={oldPassword}
                                onChange={handleOldPassword}
                                endAdornment={
                                    <InputAdornment position="end">
                                        <IconButton
                                            onClick={handleShowOldPassword}
                                        >
                                            {showOldPassword ? <VisibilityOff /> : <Visibility />}
                                        </IconButton>
                                    </InputAdornment>
                                }
                            />
                        </FormControl>
                    </Grid>
                    <Grid item xs={12}>
                        <FormControl sx={{ m: 1, width: '25ch' }} variant="standard">
                            <InputLabel>New Password</InputLabel>
                                <Input
                                type={showNewPassword ? 'text' : 'password'}
                                value={newPassword}
                                onChange={handleNewPassword}
                                endAdornment={
                                    <InputAdornment position="end">
                                        <IconButton
                                        onClick={handleShowNewPassword}
                                        >
                                        {showNewPassword ? <VisibilityOff /> : <Visibility />}
                                        </IconButton>
                                    </InputAdornment>
                                }
                                />
                        </FormControl>
                    </Grid>
                    <Grid item xs={12}>
                        <Button
                            onClick={() => {
                                if (handlePasswordChange()) {
                                    setSuccessfulBar(true);
                                } else{
                                    setErrorBar(true);
                                }
                            }}
                            variant="outlined"
                        >
                            Save Password
                        </Button>
                        <Snackbar
                            open={successfulBar}
                            autoHideDuration={6000}
                            onClose={handleClose}
                            message={'Password Saved'}
                        />
                        <Snackbar
                            open={errorBar}
                            autoHideDuration={6000}
                            onClose={handleClose}
                            message={'New Password must at least be 1 character long'}
                        />
                    </Grid>
                    <Grid item xs={12}>
                    </Grid>
                </Grid>
            </div>
        </Box>
    );
}