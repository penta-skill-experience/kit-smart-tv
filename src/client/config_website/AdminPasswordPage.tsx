import * as React from "react";
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import Button from '@mui/material/Button';

export const AdminPasswordPage = ({oldPassword, newPassword, handleOldPassword, handleNewPassword, handlePasswordChange, children} ) => {

    return (
        <Box
        sx={{
            width: 300,
            height: 300,
            }
        }
        >
            <div>
                <TextField
                    value={oldPassword}
                    label="Old Password"
                    onChange={handleOldPassword}
                />
                <TextField
                    value={newPassword}
                    label="New Password"
                    onChange={handleNewPassword}
                />
                <Button
                    onClick={handlePasswordChange}
                    variant="outlined"
                >
                    Save Password</Button>
            </div>
        </Box>
    );

}