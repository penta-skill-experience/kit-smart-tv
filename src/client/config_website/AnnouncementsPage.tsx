import * as React from "react";
import {Grid} from "@mui/material";
import TextField from '@mui/material/TextField';
import Button from "@mui/material/Button";
import DeleteIcon from "@mui/icons-material/Delete";

export const AnnouncementsPage = () => {
    const initialMailList = [];
    const [mailList, setMailList] = React.useState(initialMailList);
    const [verUser, setVerUser] = React.useState({
        mail:'',
        name:'',
    });


    const handleMailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setVerUser({...verUser, mail:event.target.value});
    };

    const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setVerUser({...verUser, name:event.target.value});
    };

    const handleAddMail = () => {
        if (verUser.name !== '' && verUser.mail !== '') {
            const newUser = {
                mail:verUser.mail,
                name:verUser.name,
            }
            setMailList(mailList.concat(newUser));
        }
        alert('Username and email have to be filled out')
    };

    const handleDeleteUser = (mail) => {
        setMailList(mailList.filter(item => item.mail !== mail));
    }


    return(
        <div>
            <Grid container spacing={2} direction="row" justifyContent="flex-start" alignItems="center">
                <Grid item xs={12}>
                    <h1>Add e-mail as verified: </h1>
                </Grid>
                <Grid item>
                    <TextField
                        label="E-Mail"
                        value={verUser.mail}
                        onChange={handleMailChange}
                    />
                </Grid>
                <Grid item>
                    <TextField
                        label="Name"
                        value={verUser.name}
                        onChange={handleNameChange}
                    />
                </Grid>
                <Grid item>
                    <Button variant="outlined" onClick={handleAddMail}>
                        Add
                    </Button>
                </Grid>
                <Grid item xs={12}>
                    <h1>Currently verified accounts:</h1>
                </Grid>
                <Grid item xs={4}>
                    Email
                </Grid>
                <Grid item xs={2}>
                    Name
                </Grid>
                <Grid item xs={6}></Grid>
                <Grid item xs={12}>
                    <ul>
                        {mailList.map(item => (
                            <Grid container spacing={2} direction="row" justifyContent="flex-start" alignItems="center">
                                <Grid item xs={4}>
                                    {item.mail}
                                </Grid>
                                <Grid item xs={1}>
                                    {item.name}
                                </Grid>
                                <Grid item xs={1}>
                                    <Button onClick={() => {handleDeleteUser(item.mail)}}>
                                        <DeleteIcon/>
                                    </Button>
                                </Grid>
                            </Grid>
                        ))}
                    </ul>
                </Grid>
            </Grid>
        </div>
    );
}