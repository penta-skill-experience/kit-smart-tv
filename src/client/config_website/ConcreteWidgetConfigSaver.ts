import {WidgetConfigSaver} from "../widget/WidgetConfigSaver";

export class ConcreteWidgetConfigSaver implements WidgetConfigSaver {

    save(rawConfig: Object): void {

        // todo write to persistence
        console.log("save widget config:");
        console.log(rawConfig);
    }
}