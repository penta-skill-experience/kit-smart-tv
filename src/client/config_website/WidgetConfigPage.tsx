import * as React from "react";
import {Button, ButtonGroup, Grid} from "@mui/material";
import MiscellaneousServicesIcon from "@mui/icons-material/MiscellaneousServices";
import Dialog from '@mui/material/Dialog';
import {ConfigComponent} from "../widget/ConfigComponent";

interface WidgetConfigPageProps {
    configComponentClass: typeof ConfigComponent;
    save: (rawConfig: Object) => void;
    rawConfig: Object;
}

interface WidgetConfigPageState {
    open: boolean;
}

export class WidgetConfigPage extends React.Component<WidgetConfigPageProps, WidgetConfigPageState> {

    private configComponentRef = React.createRef<ConfigComponent<any>>();

    constructor(props: Readonly<WidgetConfigPageProps> | WidgetConfigPageProps) {
        super(props);
        this.state = {
            open: false,
        };
    }

    private open() {
        this.setState({open: true});
    }

    private close() {
        this.setState({open: false});
    }

    private save() {
        this.props.save(this.configComponentRef.current.save());
    }

    render() {
        return <Grid item>
            <Button onClick={() => this.open()}>
                <MiscellaneousServicesIcon/>
            </Button>
            <Dialog onClose={() => this.close()} open={this.state.open} maxWidth={'xl'}
                    PaperProps={{
                        sx: {
                            padding: "24px",
                            display: "flex",
                            flexDirection: "column",
                            alignItems: "center",
                            justifyContent: "center",
                        }
                    }}>
                {
                    // @ts-ignore
                    React.createElement(this.props.configComponentClass, {
                        config: this.props.rawConfig,
                        ref: this.configComponentRef
                    }, null)
                }
                <ButtonGroup>
                    <Button variant="outlined" onClick={() => this.close()}>
                        Cancel
                    </Button>
                    <Button variant="contained" onClick={() => {
                        this.save();
                        this.close();
                    }}>
                        OK
                    </Button>
                </ButtonGroup>
            </Dialog>
        </Grid>;
    }
}
