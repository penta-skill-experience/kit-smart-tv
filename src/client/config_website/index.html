<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Admin Interface</title>
    <meta content="width=device-width, initial-scale=1" name="viewport">
    <link
            href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
            rel="stylesheet"
    />
    <style>
        #root {
            height: 100vh;
            width: 100vw;
        }
    </style>
</head>
<body>
<div id="root">
    <!-- react output will go inside here -->
</div>
</body>
</html>
