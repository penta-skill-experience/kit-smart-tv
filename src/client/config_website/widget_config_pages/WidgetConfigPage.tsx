import * as React from "react";
import {Button, Grid} from "@mui/material";
import MiscellaneousServicesIcon from "@mui/icons-material/MiscellaneousServices";
import Dialog from '@mui/material/Dialog';

export function WidgetConfigPage({content}) {
    const [open, setOpen] = React.useState(false);

    const handleOpen = () => {
        setOpen(true);
    };

    const handleClose = () => {
        setOpen(false);
    };


    return(
        <Grid item>
            <Button onClick={handleOpen}>
                <MiscellaneousServicesIcon/>
            </Button>
            <Dialog onClose={handleClose} open={open}>
                {content}
            </Dialog>
        </Grid>
    );
}