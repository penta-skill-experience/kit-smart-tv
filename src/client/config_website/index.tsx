import "./index.css";
import * as ReactDOM from "react-dom";
import * as React from "react";
import {WidgetLoader} from "../widget/WidgetLoader";
import {ConcreteWidgetConfigSaver} from "./ConcreteWidgetConfigSaver";

ReactDOM.render(
    // this is just an example to demonstrate how widgets are instantiated
    // todo: remove this.
    <div>
        <h1>Hello Admin!</h1>
        <div>
            {new WidgetLoader().getWidget("rss-feed").createConfigComponent({url: "www.example.com"}, new ConcreteWidgetConfigSaver())}
        </div>
    </div>,
    document.getElementById("root")
);