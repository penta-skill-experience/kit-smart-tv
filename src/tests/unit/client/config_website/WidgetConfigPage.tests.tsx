import {shallow, configure} from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import * as React from "react";
import toJson from "enzyme-to-json";
import {WidgetConfigPage} from "../../../../client/config_website/WidgetConfigPage";


configure({adapter: new Adapter()});

let wrapper;

describe("WidgetConfigPage tests", () => {

    beforeEach(() => {
        wrapper = shallow(
            <WidgetConfigPage
                configComponentClass={undefined}
                save={undefined}
                rawConfig={undefined}>
            </WidgetConfigPage>
        );
    });

    test("snapshot test for WidgetConfigPage", async () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

});