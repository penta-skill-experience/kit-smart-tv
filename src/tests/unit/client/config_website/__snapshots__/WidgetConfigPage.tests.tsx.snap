// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetConfigPage tests snapshot test for WidgetConfigPage 1`] = `
<ForwardRef(Grid)
  item={true}
>
  <ForwardRef(Button)
    onClick={[Function]}
  >
    <Memo(ForwardRef(MiscellaneousServicesIcon)) />
  </ForwardRef(Button)>
  <ForwardRef(Dialog)
    PaperProps={
      Object {
        "sx": Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "padding": "24px",
        },
      }
    }
    maxWidth="xl"
    onClose={[Function]}
    open={false}
  >
    <Component />
    <ForwardRef(ButtonGroup)>
      <ForwardRef(Button)
        onClick={[Function]}
        variant="outlined"
      >
        Cancel
      </ForwardRef(Button)>
      <ForwardRef(Button)
        onClick={[Function]}
        variant="contained"
      >
        OK
      </ForwardRef(Button)>
    </ForwardRef(ButtonGroup)>
  </ForwardRef(Dialog)>
</ForwardRef(Grid)>
`;
